src/net_processing.cpp

```diff
@@ -1590,6 +1590,11 @@ void static ProcessGetData(CNode* pfrom, const CChainParams& chainparams, CConnm
             if (mi != mapRelay.end()) {
                 connman->PushMessage(pfrom, msgMaker.Make(nSendFlags, NetMsgType::TX, *mi->second));
                 push = true;
+                // Once a peer requests GETDATA for a txn, we deem initial broadcast a success
+                int num = mempool.m_unbroadcast_txids.erase(inv.hash);
```

Comment:
Why is the tx not removed when it was initially sent out after 15 minutes, e.g. when it was removed from mapRelay and then relayed from the mempool.

I think this code should go down into the `if(push)` branch

Reply:
you are right, this is a mistake. thanks 🙏🏽

Reply:
Fixed. Thanks again.